src/codegen/mod.rs

```diff
@@ -3913,6 +3913,28 @@ impl CodeGenerator for ObjCInterface {
                         }
                     };
                     result.push(impl_trait);
+                    for (category_name, category_template_names) in
```

Comment:
So here's a thought, is there any reason we can't just implement this when generating code for the `ObjCInterface`?

It feels weird that stuff that would be blacklisted would implement these traits and so on. It should also avoid the weird lookup, right?

As in, instead of keeping the `category` as a `String` in the interface, you would keep the `ItemId`. Then when doing code generation you can do the lookup just fine.

Reply:
This is an interesting idea. I'm not a big fan of `String` for the type of `category`. I'll try this out.